### PR TITLE
Adds support for relative_repo_path for the docker rsync distributor

### PR DESCRIPTION
**Pull request by dkliban**
re #1887
https://pulp.plan.io/issues/1887

### CODE REVIEW

File: plugins/pulp_docker/plugins/distributors/configuration.py [hunk 1]
```diff
@@ -232,6 +232,24 @@ def get_repo_relative_path(repo, config):
...
+    distributor.
+
+    :param repo: repository to get relative path for
+    :type  repo: pulp.plugins.model.Repository
+    :param config: configuration instance for the repository
+    :type  config: pulp.plugins.config.PluginCallConfiguration or dict
+    :return: relative path for the repository
+    :rtype:  str
+    """
+    repo_relative_path = config.get('repo_relative_path', None)
```
Comment (seandst) [inline plugins/pulp_docker/plugins/distributors/configuration.py]: `repo_relative_path = config.get('repo_relative_path', repo.id)`?

Comment (bmbouter) [inline plugins/pulp_docker/plugins/distributors/configuration.py]: Yes a default of `repo.id` looks right.

Comment (bmbouter) [inline plugins/pulp_docker/plugins/distributors/configuration.py]: Or is `repo_id`?

File: plugins/pulp_docker/plugins/distributors/configuration.py [hunk 2]
```diff
@@ -232,6 +232,24 @@ def get_repo_relative_path(repo, config):
...
+def get_remote_repo_relative_path(repo, config):
+    """
+    Get the configured relative path for the given repository. This method is used by rsync
+    distributor.
+
+    :param repo: repository to get relative path for
+    :type  repo: pulp.plugins.model.Repository
+    :param config: configuration instance for the repository
+    :type  config: pulp.plugins.config.PluginCallConfiguration or dict
+    :return: relative path for the repository
```
Comment (bmbouter) [inline plugins/pulp_docker/plugins/distributors/configuration.py]: I usually add a line of whitespace separating input args from return values

File: plugins/pulp_docker/plugins/distributors/configuration.py [hunk 3]
```diff
@@ -232,6 +232,24 @@ def get_repo_relative_path(repo, config):
     return repo.id
 
 
+def get_remote_repo_relative_path(repo, config):
+    """
+    Get the configured relative path for the given repository. This method is used by rsync
```
Comment (bmbouter) [inline plugins/pulp_docker/plugins/distributors/configuration.py]: This should mention the default return behavior.

File: plugins/pulp_docker/plugins/distributors/configuration.py (hunk 1 above)
Comment (dkliban) [inline plugins/pulp_docker/plugins/distributors/configuration.py]: I updated it just use repo.id as the default. It's a repository mongoengine model. repo.id is correct.

**Comment by bmbouter** [comment]
ok test

**Comment by bmbouter** [comment]
@dkliban All comments have been addressed thanks. I also tested it with the new test from the test plan and that passed also. I also ran the tests locally which passed. Let's see if the tests pass in Jenkins, but I'm adding the LGTM now because I think it's OK to merge.

**Pull request merged**